### DISCUSSION TITLE
Automatically add some as an author if they make an edit to a post

### DISCUSSION THREAD

**Discussion by stevejburge** [publishpress/PublishPress-Authors#1225] [Feature Requests]
This is a good idea from a customer:

> Automatically add some as an author if they make an edit to a post

Essentially, this is would allow anyone who contributes to a post to automatically get credit.

**Comment by stevejburge** [comment]
@muhammedmagdi Here's my thinking here.

If this settings is enabled, anyone who has created a revision for a post is added as an Author.

If a user is stored in wp_posts as the author of a revision, they are added.

<img width="843" alt="Screen Shot 2020-08-31 at 1 06 11 PM" src="https://user-images.githubusercontent.com/3868028/91746585-d5035100-eb8a-11ea-899e-33f840626a40.png">

![Screen Shot 2020-08-31 at 1 07 11 PM](https://user-images.githubusercontent.com/3868028/91746728-05e38600-eb8b-11ea-8752-7cb7b3b23e59.jpg)

**Comment by ghost** [comment]
So, just to make sure I understand correctly, if I am the current post author of a specific post and John had created a revision for it, should he be added automatically to the post authors of this post. So when I open this post, I should see 2 authors of it, me and him? What if I want to delete him, every time the automatic trigger function will add him.

**Comment by stevejburge** [comment]
@muhammedmagdi Yes, 100% correct
"John had created a revision for it, should he be added automatically to the post authors of this post."

Can you explain the delete question? I think it should be possible to manually remove these authors

**Comment by ghost** [comment]
If I deleted John and the trigger function ( that adds an author if he has published a revision ) added him again. So when should this trigger function work? It will check every time for the users that have published any revision and add them.

**Comment by stevejburge** [comment]
Good questions @muhammedmagdi. Would it be easier in the first version if they can not be removed?

Or should the trigger run every time a revision is saved?

**Comment by ghost** [comment]
I am not sure, I am thinking for a way to make John added automatically but if the admin deleted him, this function trigger execution shouldn't work any more at this post. Just my thoughts and still thinking about it.

**Comment by stevejburge** [comment]
@muhammedmagdi I'm OK with moving on to the next task and coming back to this. Would that help?

**Comment by ghost** [comment]
That would be great. Just please let me know the next task.
